### PR TITLE
`qml.Qubitization` template

### PR DESCRIPTION
**Pull request by KetpuntoG**
Adding `qml.Qubitization` operator.

### CODE REVIEW

**Comment by KetpuntoG** [comment]
[sc - 61135]

File: patata.py
Comment (trbromley) [inline patata.py:1]: Test file?

File: pennylane/templates/subroutines/qubitization.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        if terms[0][i] >= 0:
+            new_unitaries.append(terms[1][i])
+        else:
+            new_unitaries.append(terms[1][i] @ qml.GlobalPhase(np.pi))
+
+    return qml.math.abs(hamiltonian.terms()[0]), new_unitaries
+
+
+class Qubitization(Operation):
+    r"""Applies the Qubitization operator.
```
Comment (trbromley) [inline pennylane/templates/subroutines/qubitization.py:49]: Thanks @KetpuntoG! Could we provide clickable links to `StatePrep` and `Select` so users understand the decomposition? E.g., at first I was confused about what the `control` wires were for here, until I remembered about `Select`.

File: patata.py
Comment (KetpuntoG) [reply patata.py:1]: Oops, yes! I'll delete it, thanks!

File: pennylane/templates/subroutines/qubitization.py (hunk 1 above)
Comment (KetpuntoG) [reply pennylane/templates/subroutines/qubitization.py:49]: Yes, we chose that name because of select. If you think there is a more appropriate name we can modify it.

File: pennylane/templates/subroutines/__init__.py
Comment (trbromley) [inline pennylane/templates/subroutines/__init__.py:1]: Do we also plan to have `Qubitization` show up here: https://xanaduai-pennylane--5500.com.readthedocs.build/en/5500/introduction/templates.html

**Review by soranjh — CHANGES_REQUESTED** [review]
Thanks @KetpuntoG, looks good overall. Left some comments and suggestions.

File: tests/templates/test_subroutines/test_qubitization.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+        jac_fn = jax.jacobian(qnode)
+        if use_jit:
+            jac_fn = jax.jit(jac_fn)
+
+        jac = jac_fn(params)
+        assert jac.shape == (4,)
+        assert np.allclose(jac, self.exp_grad, atol=0.01)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize("shots", [None])
```
[the remark this answers is not in the document]
Comment (KetpuntoG) [reply tests/templates/test_subroutines/test_qubitization.py:241]: if we put finite shots, the bug we have seen with the differentiability of `op @ op1(theta)` appears:

https://xanaduhq.slack.com/archives/C06DX049G49/p1713548349157569

Should I put a "TODO" there?

File: tests/templates/test_subroutines/test_qubitization.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    @pytest.mark.parametrize(
+        "use_jit , shots",
+        ((False, None), (True, None), (False, 50000)),
+    )  # TODO: (True, 50000) fails, Currently jax.jit on jax.grad does not work with AmplitudeEmbedding
+    def test_qnode_jax(self, shots, use_jit):
+        """ "Test that the QNode executes and is differentiable with JAX. The shots
+        argument controls whether autodiff or parameter-shift gradients are used."""
+        import jax
+
+        jax.config.update("jax_enable_x64", True)
```
[the remark this answers is not in the document]
Comment (soranjh) [reply tests/templates/test_subroutines/test_qubitization.py:222]: Correct.

File: tests/templates/test_subroutines/test_qubitization.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+    new_coeffs, new_unitaries = _positive_coeffs_hamiltonian(hamiltonian)
+
+    assert np.allclose(new_coeffs, np.abs(hamiltonian.terms()[0]))
+
+    for i, unitary in enumerate(new_unitaries):
+        assert qml.equal(expected_unitaries[i], unitary)
+
+
+def test_template_definition():
```
[the remark this answers is not in the document]
Comment (KetpuntoG) [reply tests/templates/test_subroutines/test_qubitization.py:71]: My conclusion here is that actually Qubitization does not generate that operator as such. It is an interpretation in a different subspace. So we're not going to get that matrix understood in the same way. For that reason, I check that the result is correct from the definition (eq 65). I collapse the value into |0> and check that the amplitude is E_k / lambda. The reason I choose Z gates is because it is easy to check this definition. Otherwise, finding the eigenvectors gets complicated. But I can think of a more complex example

File: pennylane/templates/subroutines/qubitization.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+    which can be implemented with a quantum walk operator that takes a Hamiltonian as input and generates:
+
+    .. math::
+        Q = (2|0\rangle\langle 0| - I) \text{Prep}_{\mathcal{H}}^{\dagger} \text{Sel}_{\mathcal{H}} \text{Prep}_{\mathcal{H}}.
+
+    .. seealso:: :class:`~.StatePrep` and :class:`~.Select`.
+
+    Args:
+        hamiltonian (.Hamiltonian): The Hamiltonian to be qubitized.
```
[the remark this answers is not in the document]
Comment (KetpuntoG) [reply pennylane/templates/subroutines/qubitization.py:60]: Good point, an LCU

**Review by Jaybsoni — COMMENTED** [review]
Looks very good! Just a few comments and clarifications

File: doc/releases/changelog-dev.md [hunk 6]
```diff
@@ -71,6 +71,37 @@
...
+  When applied in conjunction with QPE, allows computing the eigenvalue of an eigenvector of the Hamiltonian.
+  [(#5500)](https://github.com/PennyLaneAI/pennylane/pull/5500)
+
+  ```python
+  H = qml.dot([0.1, 0.3, -0.3], [qml.Z(0), qml.Z(1), qml.Z(0) @ qml.Z(2)])
+
+  @qml.qnode(qml.device("default.qubit"))
+  def circuit():
+
+    # initiate the eigenvector
```
Comment (soranjh) [inline doc/releases/changelog-dev.md:85]: Please use 4 spaces for indentation.

File: tests/templates/test_subroutines/test_qubitization.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (KetpuntoG) [reply tests/templates/test_subroutines/test_qubitization.py:241]: They are aware, there is a long discussion [here](https://xanaduhq.slack.com/archives/C06DX049G49/p1713548349157569)

it does not seem an easy issue

File: tests/templates/test_subroutines/test_qubitization.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (soranjh) [reply tests/templates/test_subroutines/test_qubitization.py:71]: I am still not so sure about this test. Maybe you can compare the matrices for `Q` and `R.Prep\dagger.Sel.Prep`. If so, then the test name should be changed to something like test_qubitization_matrix.

File: doc/releases/changelog-dev.md [hunk 7]
```diff
@@ -116,6 +147,7 @@
    -1.0 * X(1) @ Y(0),
    -1.0 * Y(1) @ X(0),
    -1.0 * Y(1) @ Y(0)]
+
```
Comment (KetpuntoG) [inline doc/releases/changelog-dev.md:150]:
```suggestion
```

File: doc/releases/changelog-dev.md [hunk 8]
```diff
@@ -224,6 +255,7 @@
   >>> op.error(method="commutator")
   SpectralNormError(6.166666666666668e-06)
   ```
+
```
Comment (KetpuntoG) [inline doc/releases/changelog-dev.md:258]:
```suggestion
```

File: pennylane/templates/subroutines/qubitization.py [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+        list(float), list(.Operation): The coefficients and unitaries of the transformed Hamiltonian.
+    """
+
+    new_unitaries = []
+
+    coeffs, ops = hamiltonian.terms()
+
+    for i, coeff in enumerate(coeffs):
+        angle = np.pi * (0.5 * (1 - qml.math.sign(coeff)))
+        new_unitaries.append(ops[i] @ qml.GlobalPhase(angle, wires=ops[i].wires))
```
[the remark this answers is not in the document]
Comment (KetpuntoG) [reply pennylane/templates/subroutines/qubitization.py:39]: Nice!

File: pennylane/templates/subroutines/qubitization.py [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+        eigenvalue: 0.7
+    """
+
+    def __init__(self, hamiltonian, control, id=None):
+        wires = hamiltonian.wires + qml.wires.Wires(control)
+
+        self._hyperparameters = {
+            "hamiltonian": hamiltonian,
+            "control": control,
+        }
```
[the remark this answers is not in the document]
Comment (KetpuntoG) [reply pennylane/templates/subroutines/qubitization.py:97]: Inherits the warnings from `qml.Select`

File: tests/templates/test_subroutines/test_qubitization.py [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+
+    H2 = qml.Hamiltonian(coeffs, ops)
+    matrix_H2 = qml.matrix(qml.Qubitization(H2, control=[3, 4]), wire_order=[3, 4, 0, 1, 2])
+
+    assert np.allclose(matrix_H1, matrix_H2)
+
+
+@pytest.mark.xfail(reason="see https://github.com/PennyLaneAI/pennylane/issues/5507")
+@pytest.mark.usefixtures("use_legacy_and_new_opmath")
+def test_legacy_new_opmath_diff():
```
Comment (soranjh) [inline tests/templates/test_subroutines/test_qubitization.py:100]: Why this is not inside TestDifferentiability?

File: pennylane/templates/subroutines/qubitization.py [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+        decomp_ops.append(qml.Select(unitaries, control=control))
+        decomp_ops.append(
+            qml.adjoint(
+                qml.AmplitudeEmbedding(
+                    qml.math.sqrt(coeffs), normalize=True, pad_with=0, wires=control
+                )
+            )
+        )
+
+        decomp_ops.append(qml.FlipSign(0, wires=control))
```
[the remark this answers is not in the document]
Comment (KetpuntoG) [reply pennylane/templates/subroutines/qubitization.py:144]: It is kind of weird because Reflection is expecting an operator U but in this case is the identity. A possible equivalent
circuit could be this:

```python
        decomp_ops.append(qml.Reflection(qml.prod(*[qml.Identity(wire) for wire in control])))

```



Do you want me to update it?

File: tests/templates/test_subroutines/test_qubitization.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (soranjh) [reply tests/templates/test_subroutines/test_qubitization.py:241]: Yes, let's put a TODO.

File: tests/templates/test_subroutines/test_qubitization.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (KetpuntoG) [reply tests/templates/test_subroutines/test_qubitization.py:100]: I cannot make it work 🤔

File: tests/templates/test_subroutines/test_qubitization.py [hunk 13]
```diff
@@ -0,0 +1,274 @@
...
+        qnode = qml.QNode(self.circuit, dev, interface="torch", diff_method=diff_method)
+
+        params = torch.tensor(self.params, requires_grad=True)
+        jac = torch.autograd.functional.jacobian(qnode, params)
+        assert qml.math.shape(jac) == (4,)
+        assert qml.math.allclose(jac, self.exp_grad, atol=0.01)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("shots", [None, 50000])
+    @pytest.mark.xfail(reason="tf gradient doesn't seem to be working, returns ()")
```
Comment (soranjh) [inline tests/templates/test_subroutines/test_qubitization.py:258]: Is this because of a bug in other operators?

File: tests/templates/test_subroutines/test_qubitization.py (hunk 11 above)
Comment (KetpuntoG) [reply tests/templates/test_subroutines/test_qubitization.py:100]: You are right 👍

File: tests/templates/test_subroutines/test_qubitization.py (hunk 13 above)
Comment (Jaybsoni) [reply tests/templates/test_subroutines/test_qubitization.py:238]: Ive seen it in a few operators. Also, I don't think we are making an effort to support that in the broader PennyLane, so should be fine. Maybe we can confirm with @trbromley

Comment (KetpuntoG) [reply tests/templates/test_subroutines/test_qubitization.py:238]: Indeed, this is not a particular error of this PR, it is a test that I have seen that is always put with the xfail

File: pennylane/templates/subroutines/qubitization.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (Jaybsoni) [reply pennylane/templates/subroutines/qubitization.py:145]: Identity can take multiple wires, so it should just be: 

```
decomp_ops.append(qml.Reflection(qml.Identity(control)))  # would we need to specify reflection wires ..? 
```

[the remark this answers is not in the document]
Comment (Jaybsoni) [reply pennylane/templates/subroutines/qubitization.py:145]: Give it a try, I think it's better if we can use the tools we built and keep it modular. Unless you think there is a problem with performance.

File: tests/templates/test_subroutines/test_qubitization.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (soranjh) [reply tests/templates/test_subroutines/test_qubitization.py:71]: Why not just add the example in the docs as a test? My problem with the current test is that it checks a very special case of a diagonal Hamiltonian. Also, I argue that it does not really checks Eq 65 as it claims.

File: pennylane/templates/subroutines/qubitization.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (albi3ro) [reply pennylane/templates/subroutines/qubitization.py:98]: That hasn't been added yet, so no need to worry.  Inheriting from `SymbolicOp` would eliminate any of those concerns, as well as generally help this operator adhere to the stated interface.

For example, as is the operator also needs to provide custom definitions for:
*  `_flatten`
* `_unflatten`
* `queue`
* `data`
* `data.setter`
* `__copy__`
* `hash`

and potentially others. Inheriting from `SymbolicOp` would eliminate a lot of that work.

**Review by albi3ro — COMMENTED** [review]
Mind adding a test running an instance through `qml.ops.functions.assert_valid`?

File: pennylane/templates/subroutines/qubitization.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (KetpuntoG) [reply pennylane/templates/subroutines/qubitization.py]: It worked, thanks!

File: pennylane/templates/subroutines/qubitization.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (albi3ro) [reply pennylane/templates/subroutines/qubitization.py]: Also `map_wires`. Just remembered that that one needs to get overwritten on nested ops too.

[the remark this answers is not in the document]
Comment (KetpuntoG) [reply pennylane/templates/subroutines/qubitization.py]: Thanks @albi3ro, In this case the operator does not fit with `SymbolicOp`. But I added the necessary methods to pass the test "qml.ops.functions.assert_valid". Should I also add the one you comment? Or if I already pass that test is it correct?

File: tests/templates/test_subroutines/test_qubitization.py [hunk 14]
```diff
@@ -0,0 +1,308 @@
...
+            jac_fn = jax.jit(jac_fn)
+
+        jac = jac_fn(params)
+        assert jac.shape == (4,)
+        assert np.allclose(jac, self.exp_grad, atol=0.01)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize(
+        "shots", [None]
+    )  # TODO: finite shots fails because products are not currently differentiable in this case.
```
Comment (soranjh) [inline tests/templates/test_subroutines/test_qubitization.py:254]: You mean because Prod is not differentiable?

File: tests/templates/test_subroutines/test_qubitization.py [hunk 15]
```diff
@@ -0,0 +1,308 @@
...
+@pytest.mark.parametrize(
+    "hamiltonian",
+    [
+        qml.dot([0.2, -0.5, 0.3], [qml.Y(0) @ qml.X(1), qml.Z(1), qml.X(0) @ qml.Z(2)]),
+        qml.dot([0.3, -0.5, 0.3], [qml.Z(0) @ qml.X(1), qml.X(1), qml.X(0) @ qml.Y(2)]),
+        qml.dot([0.4, -0.5, -0.3], [qml.Z(0) @ qml.X(2), qml.Y(0), qml.X(1) @ qml.Z(2)]),
+    ],
+)
+def test_operator_definition_qpe(hamiltonian):
+    """Tests Check that we can obtain the eigenvalues of H by applying the Qubitization Operator."""
```
Comment (soranjh) [inline tests/templates/test_subroutines/test_qubitization.py:80]:
```suggestion
    """Tests that Qubitization can be used in QPE to obtain the eigenvalues of a Hamiltonian."""
```

File: tests/templates/test_subroutines/test_qubitization.py [hunk 16]
```diff
@@ -0,0 +1,308 @@
...
+
+        # initial state
+        qml.RX(theta[2], wires=0)
+        qml.CRY(theta[3], wires=[0, 2])
+        qml.RY(theta[0], wires=2)
+        qml.CRY(theta[4], wires=[1, 2])
+        qml.RX(theta[1], wires=1)
+        qml.CRX(theta[2], wires=[2, 0])
+
+        # apply QPE (used iterative qpe here)
```
Comment (soranjh) [inline tests/templates/test_subroutines/test_qubitization.py:95]: Remove this, it is clear you are applying iterative_qpe.

File: tests/templates/test_subroutines/test_qubitization.py [hunk 17]
```diff
@@ -0,0 +1,308 @@
...
+        measurements = qml.iterative_qpe(
+            qml.Qubitization(hamiltonian, control=[3, 4]), ancilla=5, iters=8
+        )
+
+        return qml.probs(op=measurements)
+
+    theta = np.array([0.1, 0.2, 0.3, 0.4, 0.5]) * 100
```
Comment (soranjh) [inline tests/templates/test_subroutines/test_qubitization.py:102]: Why `* 100`?

File: tests/templates/test_subroutines/test_qubitization.py [hunk 18]
```diff
@@ -0,0 +1,308 @@
...
+
+    peaks, _ = find_peaks(circuit(theta))
+
+    # Calculates the eigenvalues from the obtained output
+    lamb = sum([abs(c) for c in hamiltonian.terms()[0]])
+    estimated_eigenvalues = lamb * np.cos(2 * np.pi * peaks / 2**8)
+
+    assert np.allclose(np.sort(estimated_eigenvalues), qml.eigvals(hamiltonian), atol=0.1)
```
Comment (soranjh) [inline tests/templates/test_subroutines/test_qubitization.py:110]: Why `atol` is so large?

File: doc/releases/changelog-dev.md [hunk 19]
```diff
@@ -114,7 +114,37 @@
...
+  When applied in conjunction with QPE, allows computing the eigenvalue of an eigenvector of the Hamiltonian.
+  [(#5500)](https://github.com/PennyLaneAI/pennylane/pull/5500)
+
+  ```python
+  H = qml.dot([0.1, 0.3, -0.3], [qml.Z(0), qml.Z(1), qml.Z(0) @ qml.Z(2)])
+
+  @qml.qnode(qml.device("default.qubit"))
+  def circuit():
+
+      # initiate the eigenvector
```
Comment (DSGuala) [inline doc/releases/changelog-dev.md:128]:
```suggestion
      # initialize the eigenvector
```

File: doc/releases/changelog-dev.md [hunk 20]
```diff
@@ -114,7 +114,37 @@
...
+      # apply QPE (used iterative qpe here)
+      measurements = qml.iterative_qpe(
+                    qml.Qubitization(H, control = [3,4]), ancilla = 5, iters = 3
+                    )
+      return qml.probs(op = measurements)
+  
+  output = circuit()
+  
+  # post-processing 
+  lamb = sum([abs(c) for c in H.terms()[0]])
```
Comment (DSGuala) [inline doc/releases/changelog-dev.md:140]: `lamb` is some form of normalization constant?

File: pennylane/templates/subroutines/qubitization.py [hunk 21]
```diff
@@ -0,0 +1,160 @@
...
+        angle = np.pi * (0.5 * (1 - qml.math.sign(coeff)))
+        new_unitaries.append(op @ qml.GlobalPhase(angle, wires=op.wires))
+
+    return qml.math.abs(coeffs), new_unitaries
+
+
+class Qubitization(Operation):
+    r"""Applies the `Qubitization <https://arxiv.org/abs/2204.11890>`__ operator.
+
+    This operator encodes a Hamiltonian written as a linear combination of unitaries into unitary operator.
```
Comment (DSGuala) [inline pennylane/templates/subroutines/qubitization.py:47]:
```suggestion
    This operator encodes a Hamiltonian, written as a linear combination of unitaries, into a unitary operator.
```

**Review by DSGuala — COMMENTED** [review]
Quick review: looks good to me in general. I left some suggestions on minor changes to the docs.

Will try to take another look and test it before the feature freeze. But no blockers from my end.

File: doc/releases/changelog-dev.md (hunk 20 above)
Comment (KetpuntoG) [reply doc/releases/changelog-dev.md:140]: yes, It is lambda, taken from the paper

File: tests/templates/test_subroutines/test_qubitization.py (hunk 17 above)
Comment (KetpuntoG) [reply tests/templates/test_subroutines/test_qubitization.py:102]: In order to make it more random, (>> 2pi ) It can be remove

File: tests/templates/test_subroutines/test_qubitization.py (hunk 18 above)
Comment (KetpuntoG) [reply tests/templates/test_subroutines/test_qubitization.py:110]: it is an approximation. Similar to the QPE test

File: pennylane/templates/subroutines/qubitization.py [hunk 22]
```diff
@@ -0,0 +1,177 @@
...
+
+    .. math::
+        Q = (2|0\rangle\langle 0| - I) \text{Prep}_{\mathcal{H}}^{\dagger} \text{Sel}_{\mathcal{H}} \text{Prep}_{\mathcal{H}}.
+
+
+
+    .. seealso:: :class:`~.AmplitudeEmbedding` and :class:`~.Select`.
+
+    Args:
+        hamiltonian (Union[.Hamiltonian, .Sum, .SProd, .LinearCombination]): The Hamiltonian written as a linear combination of unitaries.
```
Comment (soranjh) [inline pennylane/templates/subroutines/qubitization.py:59]: Just double-checking, the template really supports all `Hamiltonian`, `Sum`, `SProd`, `LinearCombination`? `Prod` should be there too?

**Review by soranjh — APPROVED** [review]
Looks good to go.

File: pennylane/templates/subroutines/qubitization.py (hunk 22 above)
Comment (KetpuntoG) [reply pennylane/templates/subroutines/qubitization.py:59]: Checked and added Prod ✅
 (Differentiation is still pending on the issue we found)

File: pennylane/templates/subroutines/qubitization.py [hunk 23]
```diff
@@ -0,0 +1,177 @@
...
+        decomp_ops.append(qml.Select(unitaries, control=control))
+        decomp_ops.append(
+            qml.adjoint(
+                qml.AmplitudeEmbedding(
+                    qml.math.sqrt(coeffs), normalize=True, pad_with=0, wires=control
+                )
+            )
+        )
+
+        decomp_ops.append(qml.Reflection(qml.Identity(control)))
```
Comment (Jaybsoni) [inline pennylane/templates/subroutines/qubitization.py:175]: 😍

File: tests/templates/test_subroutines/test_qubitization.py [hunk 24]
```diff
@@ -0,0 +1,335 @@
...
+
+        params = tf.Variable(self.params)
+        with tf.GradientTape() as tape:
+            res = qnode(params)
+
+        jac = tape.gradient(res, params)
+        assert qml.math.shape(jac) == (4,)
+        assert qml.math.allclose(res, self.exp_grad, atol=0.001)
+
+    @pytest.mark.xfail(reason="see https://github.com/PennyLaneAI/pennylane/issues/5507")
```
Comment (Jaybsoni) [inline tests/templates/test_subroutines/test_qubitization.py:290]: I am happy with merging this as is 👍🏼, we can make a shortcut ticket to fix this once the Sum has better grad support. We could also make a separate PR to do the work-around 👍🏼

File: tests/templates/test_subroutines/test_qubitization.py [hunk 25]
```diff
@@ -0,0 +1,335 @@
...
+    assert orig_op.hyperparameters["control"] is not copy_op.hyperparameters["control"]
+
+
+def test_map_wires():
+    """Test that a Qubitization operator can be mapped to a different wire mapping."""
+
+    H = qml.dot([1.0, 2.0], [qml.PauliX(0), qml.PauliZ(1)])
+
+    op = qml.Qubitization(H, control=[2, 3])
+    op2 = op.map_wires({0: 5, 1: 6, 2: 7, 3: 8})
```
Comment (Jaybsoni) [inline tests/templates/test_subroutines/test_qubitization.py:333]: What is the expected behaviour for Hamiltonian? do we map its wires as a side effect of this?

**Review by Jaybsoni — APPROVED** [review]
Great work with this one! Happy to approve, just a few questions to clarify!

File: tests/templates/test_subroutines/test_qubitization.py (hunk 25 above)
Comment (KetpuntoG) [reply tests/templates/test_subroutines/test_qubitization.py:333]: The Hamiltonian is not changed (I based my decision on `TrotterProduct`)

File: tests/templates/test_subroutines/test_qubitization.py (hunk 24 above)
Comment (KetpuntoG) [reply tests/templates/test_subroutines/test_qubitization.py:290]: Added a note in the issue :)

File: pennylane/templates/subroutines/qubitization.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (KetpuntoG) [reply pennylane/templates/subroutines/qubitization.py]: [Added]
